### DISCUSSION TITLE
How do composite custom types look like in PostgREST?

### DISCUSSION THREAD

**Discussion by dreinon** [PostgREST/postgrest#1990] [Q&A] [answered]
I would like to know how user-defined [composite custom types](https://www.postgresql.org/docs/9.2/sql-createtype.html) look like in a select query and what's the format I have to follow to do an insert in a table a column of which is a composite type.

Thanks 🙂

**Comment by wolfgangwalther** [comment, marked as answer]
> I would like to know how user-defined [composite custom types](https://www.postgresql.org/docs/9.2/sql-createtype.html) look like in a select query 

Have a look at this old thread, there's an example: https://github.com/PostgREST/postgrest/issues/837#issuecomment-291567909

(Ignore the whole bug thing - it's working as expected)

> and what's the format I have to follow to do an insert in a table a column of which is a composite type.

See https://github.com/PostgREST/postgrest/issues/837#issuecomment-360646039. It's a bit older, but still valid. You can either make it a string according to postgresql text->composite type conversion or make it an object, which will be parsed into the composite type.

**Reply by dreinon** [reply]
Thanks!
